docs/LPC_Additional_LPC_Cycles.md

```diff
@@ -0,0 +1,248 @@
++  First, one internal signal has been added:
+
+```verilog
+reg skipCycle;               // 1 -indicates that this cycle is not I/O or TPM cycle, 0 - indicates I/O or TPM cycle
+```
++ On the main state machine (FSM) supporting LPC cycles - states `LPC_ST_IDLE`
+and` LPC_ST_START` were modified:
+
+```verilog
+        case(current_state_o)
+            `LPC_ST_IDLE:
+             begin
+                 skipCycle <= 1'b0;
+                 if (nrst_i == 1'b0) fsm_next_state <= `LPC_ST_IDLE;
+                 else if ((lframe_i == 1'b0) && (lad_bus == 4'h0)) fsm_next_state <= `LPC_ST_START;
+             end
+             `LPC_ST_START:
+              begin
+                  if ((lframe_i == 1'b0) && (lad_bus == 4'h0)) fsm_next_state <= `LPC_ST_START;
+                  else if ((lframe_i == 1'b1) && (lad_bus != 4'h0) && (lad_bus != 4'h2)) skipCycle = 1'b1;
+                  else if ((lframe_i == 1'b1) && (lad_bus == 4'h0)) fsm_next_state <= `LPC_ST_CYCTYPE_RD;
+                  else if ((lframe_i == 1'b1) && (lad_bus == 4'h2)) fsm_next_state <= `LPC_ST_CYCTYPE_WR;                
+              end
+```
+When in the `LPC_ST_START` state, the LAD bus does not have a CYCTYPE meaning I / O`
+or `Memory` cycles, the `skipCycle` signal becomes High.
+
++ In the rest of the code, the value of the `lpc_en_o` signal is worked out already
+using the signal state of the` skipCycle`
+
+```verilog
+   assign lpc_en_o = ((!skipCycle)&&(sync_en == 1'b1)) ? 1'h1 :
+                      ((!skipCycle)&&(tar_F == 1'b1 )) ? 1'h1 :
+                      (lframe_i == 1'b0 ) ? 1'h0 :
+                      ((!skipCycle)&&(rd_data_en[0] == 1'b1)) ? 1'b1 :
+                      ((!skipCycle)&&(rd_data_en[1] == 1'b1)) ? 1'b1 :
+                      1'h0;
+```
+This causes LPC cycle data not to be recorded on the output bus for cycles other
+ than `I/O` and` TPM`.
+
+## Changes in the `test bench` implementation regarding the handling of `Memory R/W` cycles
+
++ In the test bench, firstly we added `memory_cycle_sig` to the list of internal
+  signals of the `lpc_periph_tb` module:
+ 
+```verilog
+    reg  [15:0] u_addr;    //auxiliary host addres
+    reg   [7:0] u_data;    //auxiliary host data
+    integer i, j;
+    reg memory_cycle_sig;
+```
++ Second, to the main loop sending the cycle data to module `lpc_host` we add 
+   second for loop for setting `memory_cycle_sig` signal value. In this way, the
+ `I/O` and `Memory` cycles are alternately sent:
+
+```verilog
+       for (i = 0; i <= 128; i = i + 1) begin
+          for(j = 0; j < 2; j = j + 1) begin
+            memory_cycle_sig = j; //Cycle type: Memory or I/O
+            // Perform write
+            #40  lframe_i  = 0;
+            rd_flag = 0;
+            wr_flag  = 1;
+            host_addr_i = u_addr+i;
+            host_wr_i  = u_data+i;
+            #40 lframe_i = 1;
+            #400 lframe_i = 0;
+. . .
+```
+## The results of the simulation of the `lpc-periph` module in Xilinx vivado 2019.1
```

Comment:
Header claims the simulation was done in Vivado but document shows Icarus simulation only. Also the header should be shorter, e.g. `Simulation results`, this applies to other headers too.